frontend/src/app/modules/auth/sign-in/sign-in.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+    selector: 'app-sign-in',
+    templateUrl: './sign-in.component.html',
+    styleUrls: ['./sign-in.component.sass'],
+})
+export class SignInComponent {
+    password: string = '';
+
+    email = '';
+
+    submitted = false;
+
+    form: FormGroup = new FormGroup({
+        email: new FormControl(this.email),
+        password: new FormControl(this.password),
+    });
+
+    constructor(private formBuilder: FormBuilder) {}
+
+    getEmailErrorMessage() {
+        if (this.form.controls['email'].hasError('required')) {
+            return 'You must enter a value';
+        }
+
+        if (this.form.controls['email'].hasError('email')) {
+            return 'Not a valid email';
+        }
+
+        return this.form.controls['email'].hasError('maxlength') ? 'Can not be more than 30 characters' : '';
```

Comment:
Hmmm, what about creating a function and dictionary of validation rules?)
{ 'required', 'You muse enter a value' }
And just make the same

Reply:
got it, fixed